Samples/DirectMLNpuInference/main.cpp

```diff
@@ -13,11 +13,46 @@
 
 using Microsoft::WRL::ComPtr;
 
-void InitializeDirectML(ID3D12Device1** d3dDeviceOut, ID3D12CommandQueue** commandQueueOut, IDMLDevice** dmlDeviceOut) {
-    // Whether to skip adapters which support Graphics in order to target NPU for testing
-    bool forceComputeOnlyDevice = true;
-    bool forceGenericMLDevice = false;
-    
+bool TryGetProperty(IDXCoreAdapter* adapter, DXCoreAdapterProperty prop, std::string& outputValue)
+{
+    if (adapter->IsPropertySupported(prop))
+    {
+        size_t propSize;
+        THROW_IF_FAILED(adapter->GetPropertySize(prop, &propSize));
+
+        outputValue.resize(propSize);
+        THROW_IF_FAILED(adapter->GetProperty(prop, propSize, outputValue.data()));
+
+        // Trim any trailing nul characters. 
+        while (!outputValue.empty() && outputValue.back() == '\0')
+        {
+            outputValue.pop_back();
+        }
+
+        return true;
+    }
+    return false;
+}
+
+bool GetNonGraphicsAdapter(IDXCoreAdapterList* adapterList, IDXCoreAdapter** outAdapter)
+{
+    for (uint32_t i = 0, adapterCount = adapterList->GetAdapterCount(); i < adapterCount; i++)
+    {
+        ComPtr<IDXCoreAdapter> possibleAdapter;
+        THROW_IF_FAILED(adapterList->GetAdapter(static_cast<uint32_t>(i), IID_PPV_ARGS(&possibleAdapter)));
```

Comment:
`i` is already uint32_t so no need to cast

Reply:
Removed, thanks!